### PR TITLE
PT-1278 Integrate Tidepool master for hydrophone

### PR DESCRIPTION
**Pull request by CedricPerotto**

### CODE REVIEW

File: .rgignore [hunk 1]
```diff
@@ -0,0 +1 @@
+vendor
```
Comment (yvon-dblg) [inline .rgignore]: what is this file for ?
should it be `.gitignore` ?

File: .travis.yml [hunk 2]
```diff
@@ -70,7 +70,6 @@ deploy:
```
Comment (yvon-dblg) [inline .travis.yml:70]: There is still some warnings from travis:

Build config validation
- root: deprecated key sudo (The key `sudo` has no effect anymore.)
- deploy: deprecated key skip_cleanup (not supported in dpl v2, use cleanup)
- deploy: deprecated key skip_cleanup (not supported in dpl v2, use cleanup)
- root: missing os, using the default linux
- deploy: key upload-dir is not underscored, using upload_dir
- deploy: key local-dir is not underscored, using local_dir
- deploy: key upload-dir is not underscored, using upload_dir

File: Dockerfile [hunk 3]
```diff
@@ -14,19 +14,16 @@ RUN  ./build.sh
 
 CMD ["./dist/hydrophone"]
```
Comment (yvon-dblg) [inline Dockerfile:15]: Useful ?

File: hydrophone.go [hunk 4]
```diff
@@ -69,12 +69,21 @@ func main() {
...
 	if found {
 		config.ShorelineConfig.Secret = serverSecret
 		config.Api.ServerSecret = serverSecret
 	}
+
+	protocol, found := os.LookupEnv("PROTOCOL")
+	if found {
+		config.Api.Protocol = protocol
+	} else {
+		config.Api.Protocol = "https"
```
Comment (yvon-dblg) [inline hydrophone.go:85]: Will we need some certificates for the rest of the stack ?

**Review by yvon-dblg — COMMENTED** [review]
Few questions here & there.

File: .rgignore (hunk 1 above)
Comment (CedricPerotto) [reply .rgignore]: [ripgrep ](https://github.com/BurntSushi/ripgrep) ignore file (who is able to use .gitignore files).

We don't use this tool let's delete this file.

File: .travis.yml (hunk 2 above)
Comment (CedricPerotto) [reply .travis.yml:70]: Some of these warnings concerns the future version of deploy functionality ([dplv v2](https://docs.travis-ci.com/user/deployment-v2)) which is for now in beta.
We currently use [dpl v1](https://docs.travis-ci.com/user/deployment), none of the following suggestions are documented:
- upload_dir  (upload-dir is)
- cleanup (skip_cleanup is)

I will push a change on local-dir which is not the key documented in v1, and on os/sudo key.

File: Dockerfile (hunk 3 above)
Comment (CedricPerotto) [reply Dockerfile:15]: If you want to run the "development" image, yes it is.

File: hydrophone.go (hunk 4 above)
Comment (CedricPerotto) [reply hydrophone.go:85]: This env is about the protocol used by the main API (i.e. yourloops). It is only used to generate href attributes in html email templates (see getWebURL function and WebURL key in templates).

The config of http/https for the service is stored in config.Service.Scheme (loaded from TIDEPOOL_HYDROPHONE_SERVICE env).

**Review by yvon-dblg — APPROVED** [review]
LGTM

**Pull request merged**